Pyjion/pyjit.cpp

```diff
@@ -269,10 +333,115 @@ PyObject* __stdcall Jit_EvalTrace(void* state, PyFrameObject *frame) {
         // Update the jitted information for this tree node
         curNode->addr = (Py_EvalFunc)res->get_code_addr();
         curNode->jittedCode = res;
+        if (!isSpecialized) {
+            trace->Generic = curNode->addr;
+            PyjionJittedCode* pyjionCode = (PyjionJittedCode*)frame->f_code->co_extra;
+            pyjionCode->j_evalfunc = Jit_EvalGeneric;
+        }
 
         // And finally dispatch to the newly compiled code
         return Jit_EvalHelper(curNode->addr, frame);
     }
+#else
+
+    for (auto cur = trace->opt.begin(); cur != trace->opt.end(); cur++) {
+        PyObject** locals = frame->f_localsplus;
+        bool mismatch = false;
+        auto opt = *cur;
+        for (auto type = opt->types.begin(); type != opt->types.end(); type++, locals++) {
```

Comment:
No, locals will always have at least frame->f_code->co_argcount + frame->f_code->co_kwonlyargcount entries, which is how many types we'll put into the vector.